dev/com.ibm.ws.classloading/src/com/ibm/ws/classloading/internal/ThreadContextClassLoader.java

```diff
@@ -213,17 +232,41 @@ public String getKey() {
     }
 
     final boolean isFor(ClassLoader classLoader) {
-        return classLoader == appLoader;
+        return classLoader == appLoaderRef.get();
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder(super.toString());
+        sb.append(": Application ClassLoader ");
+        sb.append(appLoaderRef.get());
+        return sb.toString();
     }
 
     // This is reflectively called by Spring for GLIBC proxy classes.
     // To work properly with package private application class beans these proxies
     // must be defined with the app loader so they have proper visibility to their
     // super classes.
+    @Override
     public Class<?> publicDefineClass(String name, byte[] b, ProtectionDomain protectionDomain) {
+        ClassLoader appLoader = this.appLoaderRef.get();
         if (appLoader instanceof SpringLoader) {
             return ((SpringLoader) appLoader).publicDefineClass(name, b, protectionDomain);
         }
         return defineClass(name, b, 0, b.length, protectionDomain);
     }
+
+    /**
+     * This method is called by the classgraph open source library doing reflection on 
+     * this class
+     * 
+     * @return AppClassLoader.getClassPath() result
+     */
+    public Collection<Collection<URL>> getClassPath() {
+        ClassLoader appLoader = appLoaderRef.get();
+        if (appLoader instanceof AppClassLoader) {
+            return ((AppClassLoader) appLoader).getClassPath();
+        }
+        return null;
+    }
```

Comment:
I gather this is not related to the other changes around weak references.  Seems this reflection would not have worked before this PR.  Not asking for changes here, only trying to understand what prompted this addition.

Reply:
I had no idea what classgraph was, I thought it was some liberty specific tool, but looks like an interesting general purpose tool for runtimes.  Thanks for the explanation.  I think the comment is fine, now that I know what classgraph is.